### ISSUE TITLE
Plugin doesn't exist (build CLI-only version)

### ISSUE THREAD

**Issue by leonstafford** [WP2Static/wp2static-addon-vercel#1]
Pretty big bug. Let's fix by creating the add-on!

**User story**

As a user
I want to be able to dpeloy my site to [Vercel](https://vercel.com)
So that I can ride their platform's wave of popularity

**References**

 - [deployments API](https://vercel.com/docs/api#endpoints/deployments)
 - [upload deployment files API](https://vercel.com/docs/api#endpoints/deployments/upload-deployment-files)
 - [existing Netlify addon](https://github.com/leonstafford/wp2static-addon-netlify) *likely to be close comparison*
 - [existing Zip addon](https://github.com/leonstafford/wp2static-addon-zip)
 - [existing S3/Cloudfront addon](https://github.com/leonstafford/wp2static-addon-s3)
 
 **Expected WP-CLI behaviour**
 
  - [ ] `wp wp2static vercel` top level, with usage info
  - [ ] `wp wp2static addons toggle vercel` activates addon (can use with `wp wp2static addons list` to verify state)
  - [ ] `wp wp2static vercel options set SOME_OPTION_NAME SOME_VALUE` saves value for addon
  - [ ] `wp wp2static vercel options list` lists values for addon
  - [ ] `wp wp2static deploy` deploys wp2static's generated site to the vercel saved configuration (`wp wp2static full_workflow` generates the required files)

  
**Expected structure/other behaviour**

 - [ ] include in README file, the steps user needs to [prepare their Vercel account via CLI](https://vercel.com/docs/cli) before using this plugin
 - [ ] keep majority conf files in `./conf` to keep repo root tidy (like in [WordPressURLDetector](https://github.com/leonstafford/WordPressURLDetector)   
 - [ ] similar `composer run-script` scripts like in [WordPressURLDetector](https://github.com/leonstafford/WordPressURLDetector) 
 - [ ] target min PHP 7.4
 
 **Test coverage**
 
 - [ ] unit tests not required. nice, if a non-trivial function. integration tests, mocking Vercel API will be nice, but not required.   
 - [ ] Leon to initially manually test + add into Lokl's [CLI end to end tests using ShellSpec](https://github.com/leonstafford/lokl/blob/master/spec/lokl_spec.sh)

**Comment by leonstafford** [comment]
@szepeviktor ready for review

**Comment by szepeviktor** [comment]
Seen & read.

_Viktor_

**Comment by szepeviktor** [comment]
> keep majority conf files in ./conf to keep repo root tidy

I've never seen (this way) hidden config files.
@leonstafford Could you show me a popular (100+ stars) repository having PHPStan where the config file is not in the root?

**Comment by leonstafford** [comment]
Yep, this one, at least, I have used for one private one, too: https://github.com/leonstafford/WordPressURLDetector

**Comment by leonstafford** [comment]
Oops, sorry, not popular (yet)

**Comment by leonstafford** [comment]
<img width="1016" alt="Screen Shot 2021-07-01 at 4 51 24 am" src="https://user-images.githubusercontent.com/2066833/124019297-09080680-da28-11eb-83be-1af5e092df90.png">

Fixed it! 

May be unpopular, but I don't want to be distracted by so many conf files in every project root.

**Comment by szepeviktor** [comment]
All right. I add it to the _Customer's Wish_ list.
